src/backend/curses/pan.rs

```diff
@@ -0,0 +1,221 @@
+        pancurses::curs_set(0);
+        window.bkgd(pancurses::COLOR_PAIR(ColorStyle::Background.id() as u64));
+
+        PancursesBackend { window: window }
+    }
+
+    fn screen_size(&self) -> (usize, usize) {
+        let (x, y) = self.window.get_max_yx();
+        (x as usize, y as usize)
+    }
+
+    fn has_colors(&self) -> bool {
+        pancurses::has_colors()
+    }
+
+    fn finish(&mut self) {
+        pancurses::endwin();
+    }
+
+    fn init_color_style(&mut self, style: ColorStyle, foreground: &Color,
+                        background: &Color) {
+        pancurses::init_pair(style.id(),
+                             find_closest(foreground) as i16,
+                             find_closest(background) as i16);
+    }
+
+    fn with_color<F: FnOnce()>(&self, color: ColorStyle, f: F) {
+        // TODO: pancurses doesn't have an `attr_get` equivalent
+        // let mut current_style: pancurses::attr_t = 0;
+        // let mut current_color: i16 = 0;
+        // pancurses::attr_get(&mut current_style, &mut current_color);
+
+        let style = pancurses::COLOR_PAIR(color.id() as u64);
+        self.window.attron(style);
+        f();
+        self.window.attroff(style);
+        // pancurses::attron(current_style);
+    }
+
+    fn with_effect<F: FnOnce()>(&self, effect: Effect, f: F) {
+        let style = match effect {
+            // A_REVERSE, taken from ncurses
+            Effect::Reverse => 1 << (10 + 8u32),
+            Effect::Simple => pancurses::A_NORMAL,
+        };
+        self.window.attron(style);
+        f();
+        self.window.attroff(style);
+    }
+
+    fn clear(&self) {
+        self.window.clear();
+    }
+
+    fn refresh(&self) {
+        self.window.refresh();
+    }
+
+    fn print_at(&self, (x, y): (usize, usize), text: &str) {
+        self.window.mvaddstr(y as i32, x as i32, text);
+    }
+
+    fn poll_event(&self) -> Event {
+        // TODO: there seems to not be any indication
+        // of Ctrl/Alt/Shift in these :v
+        if let Some(ev) = self.window.getch() {
+            match ev {
+                pancurses::Input::Character(c) => Event::Char(c),
+                pancurses::Input::Unknown(i) => Event::Unknown(i),
+                // TODO: I honestly have no fucking idea what KeyCodeYes is
+                pancurses::Input::KeyCodeYes => Event::Refresh,
+                // TODO: There seems to not be a Key::Break??
+                pancurses::Input::KeyBreak => Event::Refresh,
+                pancurses::Input::KeyDown => Event::Key(Key::Down),
+                pancurses::Input::KeyUp => Event::Key(Key::Up),
+                pancurses::Input::KeyLeft => Event::Key(Key::Left),
+                pancurses::Input::KeyRight => Event::Key(Key::Right),
+                pancurses::Input::KeyHome => Event::Key(Key::Home),
+                pancurses::Input::KeyBackspace => Event::Key(Key::Backspace),
+                // TODO: there seems to not be an F0 key for some reason?
```

Comment:
Is F0 a valid key? I never saw one, and it looked like a bitmask in the ncurses source, but I can add it if it's used.